src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/Events/SpeechToTextRecognitionResultUpdatedEventArgs.cs

```diff
@@ -5,16 +5,16 @@ namespace CommunityToolkit.Maui.Media;
 /// </summary>
 public class SpeechToTextRecognitionResultUpdatedEventArgs : EventArgs
 {
-	/// <summary>
-	/// Speech recognition result
-	/// </summary>
-	public string RecognitionResult { get; }
-
 	/// <summary>
 	/// Initialize a new instance of <see cref="SpeechToTextRecognitionResultUpdatedEventArgs"/>
 	/// </summary>
 	public SpeechToTextRecognitionResultUpdatedEventArgs(string recognitionResult)
 	{
 		RecognitionResult = recognitionResult;
 	}
+
+	/// <summary>
+	/// Speech recognition result
```

Comment:
This has the same docs as the `SpeechToTextRecognitionResultCompletedEventArgs`, and `SpeechToTextStateChangedEventArgs` can't we merge this as single one? I know, the name is little different, but the arguments and meanings are the same

Reply:
I would keep them separate. The same as in Windows native implementation. They may be extended later with more properties (Choices, recognition accuracy etc)

Reply:
we can have a base type with those properties, I believe all Args will have the same properties or most of them will be the same

Reply:
You both make good points. 

I just played around a bit with this in code and I think it's best to keep the current implementation for now.

We can't create `public abstract class SpeechToTextRecognitionEventArgs` and add `public string RecognitionResult { get; }` to it because then we's have a 3rd `EventArgs`, `SpeechToTextStateChangedEventArgs`, that doesn't implement this base class. Same is true for creating `public interface ISpeechToTextRecognitionEventArgs`; we'd always have a 3rd event args that doesn't implement an abstraction which may lead to confusion for devs expecting it.

There's nothing smelly in this code. But if/when we add additional `EventArgs` to `ISpeechToText`, we'd definitely want to rediscuss this.